embassy-stm32f4/src/serial.rs

```diff
@@ -135,43 +136,52 @@ impl Serial<USART1, Stream7<DMA2>, Stream2<DMA2>> {
 
         STATE.rx_int.signal(());
     }
+}
+
+impl Uart for Serial<USART1, Stream7<DMA2>, Stream2<DMA2>> {
+    type SendFuture<'a> = impl Future<Output = Result<(), Error>> + 'a;
+    type ReceiveFuture<'a> = impl Future<Output = Result<(), Error>> + 'a;
 
     /// Sends serial data.
     ///
     /// `tx_buffer` is marked as static as per `embedded-dma` requirements.
     /// It it safe to use a buffer with a non static lifetime if memory is not
     /// reused until the future has finished.
-    pub fn send<'a, B: 'a>(&'a mut self, tx_buffer: B) -> impl Future<Output = ()> + 'a
-    where
-        B: StaticWriteBuffer<Word = u8>,
-    {
+    #[allow(mutable_transmutes)]
```

Comment:
This is super bad, but Transfer::init requires mutable, so we must.

Reply:
Yeah, it's OK. Can you move it to just the line that requires it, instead of the entire function?